gossip/privdata/pvtdataprovider.go

```diff
@@ -202,27 +202,44 @@ func (pdp *PvtdataProvider) RetrievePvtdata(pvtdataToRetrieve []*ledger.TxPvtdat
 
 	pvtdata := make(rwsetByKeys)
 
+	//If there is no private data to retrieve for the block, skip all population attempts and return
+	if len(pvtdataRetrievalInfo.eligibleMissingKeys) == 0 {
+		pdp.logger.Debugf("No eligible collection private write sets to fetch for block [%d]", pdp.blockNum)
+		retrievedPvtdata.pvtdataRetrievalInfo = pvtdataRetrievalInfo
+		retrievedPvtdata.blockPvtdata = pdp.prepareBlockPvtdata(pvtdata, pvtdataRetrievalInfo)
+		return retrievedPvtdata, nil
+	}
+
+	numEligibleMissingKeysToRetrieve := len(pvtdataRetrievalInfo.eligibleMissingKeys)
+
 	// POPULATE FROM CACHE
+	numRemainingToFetch := len(pvtdataRetrievalInfo.eligibleMissingKeys)
```

Comment:
Nit. Looks redundant given that we can use `numEligibleMissingKeysToRetrieve` in line 218. However, I like the code consistency as similar statement is used in line 229 and 242.

Reply:
done.